spec/system/editing_a_reservation_spec.rb

```diff
@@ -91,4 +91,152 @@
       expect(page).to have_content "cannot be shortened once the reservation has started"
     end
   end
+
+  describe "with a daily booking instrument" do
+    let(:today) { Time.current.beginning_of_day }
+    let(:unavailable_period_start) { today + 3.days }
+    let(:instrument) do
+      create(
+        :setup_instrument,
+        :always_available,
+        :daily_booking,
+      )
+    end
+    let!(:reservation) do
+      create(
+        :purchased_reservation,
+        user:,
+        reserve_start_at: today + 10.days,
+        reserve_end_at: today + 11.days,
+        product: instrument,
+      )
+    end
+    let!(:other_reservation) do
+      create(
+        :purchased_reservation,
+        user: create(:user),
+        reserve_start_at: today + 1.day,
+        reserve_end_at: today + 2.days,
+        product: instrument,
+      )
+    end
+
+    def move_reservation
+      visit reservations_path
+
+      expect(page).to have_content("Move Up")
+
+      click_link("Move Up")
+
+      expect(page).to have_content("Would you like to move your reservation?")
+
+      click_button("Move")
+
+      expect(page).to have_content("The reservation was moved successfully")
+
+      expect(page).to_not have_content("Move Up")
+    end
+
+    shared_examples "unavailable due to reason" do |error_key|
+      it "fail to create a new reservation in the unavailable day" do
+        new_reserve_start_at = unavailable_period_start + 1.minute
+        invalid_reservation = build(
+          :purchased_reservation,
+          user:,
+          reserve_start_at: new_reserve_start_at,
+          reserve_end_at: new_reserve_start_at + 1.day,
+          product: instrument
+        )
+
+        expect(invalid_reservation).to_not be_valid
+        expect(invalid_reservation.errors).to be_added(:base, error_key)
+      end
+    end
+
+    describe "when unavailable due to schedule rule" do
```

Comment:
I think context block fits better in this cases

https://dev.to/vasily/explaining-rspec-describe-context-in-2-minutes-264h